### ISSUE TITLE
Compile/test error on Apple M1

### ISSUE THREAD

**Issue by JakobAsslaender** [JuliaApproximation/FastTransforms.jl#155]
Hi!
I just quickly wanted to draw your attention to an error I get when trying to install FastTransforms.jl (or ApproxFun.jl) on an Apple M1 CPU natively (Julia v1.7.0-rc1). I know this platform is only Tier-3 and has still many issues, but maybe there is an easy fix for this? The issue seems to be related to BinaryProvider.jl, yet, I am posting this here since BinaryProvider.jl does not seem to maintained based on the last commit being 1.5 years old. 

Thanks a ton for looking into this!

The error message when running `test FastTransforms` message is:

```
Precompiling project...
  ✗ BinaryProvider
  0 dependencies successfully precompiled in 1 seconds (49 already precompiled)
  1 dependency errored. To see a full report either run `import Pkg; Pkg.precompile()` or load the package
     Testing Running tests...
ERROR: LoadError: InitError: UndefVarError: libfasttransforms not defined
Stacktrace:
 [1] ft_set_num_threads(n::Int64)
   @ FastTransforms ~/.julia/packages/FastTransforms/BaTvR/src/libfasttransforms.jl:13
 [2] __init__()
   @ FastTransforms ~/.julia/packages/FastTransforms/BaTvR/src/libfasttransforms.jl:18
 [3] _include_from_serialized(path::String, depmods::Vector{Any})
   @ Base ./loading.jl:768
 [4] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String)
   @ Base ./loading.jl:854
 [5] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1097
 [6] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:1013
 [7] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:997
 [8] include(fname::String)
   @ Base.MainInclude ./client.jl:451
 [9] top-level scope
   @ none:6
during initialization of module FastTransforms
in expression starting at /Users/asslaj01/.julia/packages/FastTransforms/BaTvR/test/runtests.jl:1
ERROR: Package FastTransforms errored during testing
```

**Comment by JakobAsslaender** [comment]
Thanks, @MikaelSlevinsky for the quick response and the kind support! The error I am getting again related to the BinaryProvider.jl package and the issue [#205](https://github.com/JuliaPackaging/BinaryProvider.jl/issues/205).

Here is the error message:

```
(@v1.7) pkg> build FastTransforms
    Building TimeZones ─────→ `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/b4c6460412b1db0b4f1679ab2d5ef72568a14a57/build.log`
    Building FastTransforms → `~/.julia/dev/FastTransforms/deps/build.log`
ERROR: Error building `FastTransforms`: 
┌ Warning: Platform `arm64-apple-darwin20.5.0` is not an officially supported platform
└ @ BinaryProvider ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:450
ERROR: LoadError: KeyError: key "unknown" not found
Stacktrace:
  [1] getindex
    @ ./dict.jl:481 [inlined]
  [2] parse_dl_name_version
    @ ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:511 [inlined]
  [3] detect_libgfortran_abi(libgfortran_name::String, platform::BinaryProvider.UnknownPlatform) (repeats 2 times)
    @ BinaryProvider ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:555
  [4] detect_libgfortran_abi
    @ ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:584 [inlined]
  [5] detect_compiler_abi()
    @ BinaryProvider ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:667
  [6] top-level scope
    @ ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:685
  [7] include(mod::Module, _path::String)
    @ Base ./Base.jl:420
  [8] include(x::String)
    @ BinaryProvider ~/.julia/packages/BinaryProvider/U2dKK/src/BinaryProvider.jl:1
  [9] top-level scope
    @ ~/.julia/packages/BinaryProvider/U2dKK/src/BinaryProvider.jl:12
 [10] include
    @ ./Base.jl:420 [inlined]
 [11] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
    @ Base ./loading.jl:1318
 [12] top-level scope
    @ none:1
 [13] eval
    @ ./boot.jl:373 [inlined]
 [14] eval(x::Expr)
    @ Base.MainInclude ./client.jl:453
 [15] top-level scope
    @ none:1
in expression starting at /Users/asslaj01/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:685
in expression starting at /Users/asslaj01/.julia/packages/BinaryProvider/U2dKK/src/BinaryProvider.jl:1
ERROR: LoadError: Failed to precompile BinaryProvider [b99e7846-7c00-51b0-8f62-c81ae34c0232] to /Users/asslaj01/.julia/compiled/v1.7/BinaryProvider/jl_efaTyB.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
   @ Base ./loading.jl:1466
 [3] compilecache(pkg::Base.PkgId, path::String)
   @ Base ./loading.jl:1410
 [4] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1120
 [5] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:1013
 [6] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:997
 [7] include(fname::String)
   @ Base.MainInclude ./client.jl:451
 [8] top-level scope
   @ none:5
in expression starting at /Users/asslaj01/.julia/dev/FastTransforms/deps/build.jl:1
```

**Comment by abhayap** [comment]
@MikaelSlevinsky I am interested in using FastTransforms on an Apple M1 CPU too. I currently get this error while trying to install it:
```
[057dd010] ↑ FastTransforms v0.13.1 ⇒ v0.13.2
    Building FastTransforms → `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/848570d9a053b0e270b79a9cea95b409b317bb07/build.log`
ERROR: Error building `FastTransforms`: 
┌ Warning: Platform `arm64-apple-darwin21.1.0` is not an officially supported platform
└ @ BinaryProvider ~/.julia/packages/BinaryProvider/U2dKK/src/PlatformNames.jl:450
```

**Comment by MikaelSlevinsky** [comment]
Hi, in principle this would be feasible by a small change to the build script in Julia's Yggdrasil. I'm personally hesitant because

1. I think there is an issue of lacking testing machines https://github.com/actions/virtual-environments/issues/2187
2. I don't have an M1 computer personally, so couldn't provide any nontrivial support
3. It's great that it compiles, but I'm stumped by the segmentation fault above. (I could understand an illegal instruction, but memory issues are more important and harder to debug. See point 2.).

All that said, enthusiastic pull requests from folks with the right tools would not be ignored!

**Comment by abhayap** [comment]
I sent a PR here that should work on Apple M1 using Julia v1.7: https://github.com/JuliaPackaging/Yggdrasil/pull/4161

**Comment by dlfivefifty** [comment]
This is awesome, is it time to get an M1?? (Or maybe wait til M2 at this point)

**Comment by MikaelSlevinsky** [comment]
Work bought me an M1 pro for Christmas though it won't come in for a couple more weeks. (I didn't want to mention it to falsely get people's hopes up, but yeah this is great progress @abhayap)

**Comment by abhayap** [comment]
Thanks @MikaelSlevinsky!

**Comment by MikaelSlevinsky** [comment]
Let's assume this issue is closed now with v0.13.3 on Julia v1.7
